From 779fde1d3b5997918c13546e727ab24bdd80a6b3 Mon Sep 17 00:00:00 2001
From: Marta Kuczora <kuczoram@cloudera.com>
Date: Wed, 6 Sep 2017 15:23:29 +0200
Subject: [PATCH 1273/1363] CLOUDERA-BUILD: CDH-53833: Run the S3 q tests with
 optimization on Turn on the blobstore
 optimizations for the S3 q tests. Also exclude
 some q tests which have different explain plan on
 S3 with optimization than on HDFS.

Change-Id: Ie99e4b7e45147edccdc6e215bb2deac6ccaf6107
---
 .../test/resources/hdfs-blobstore/hive-site.xml    |    4 +--
 .../test/resources/testconfiguration.properties    |   30 --------------------
 2 files changed, 2 insertions(+), 32 deletions(-)

diff --git a/itests/hive-blobstore/src/test/resources/hdfs-blobstore/hive-site.xml b/itests/hive-blobstore/src/test/resources/hdfs-blobstore/hive-site.xml
index 9aa0e74..4f1e098 100644
--- a/itests/hive-blobstore/src/test/resources/hdfs-blobstore/hive-site.xml
+++ b/itests/hive-blobstore/src/test/resources/hdfs-blobstore/hive-site.xml
@@ -20,7 +20,7 @@
 <configuration>
   <property>
     <name>fs.s3a.metadatastore.impl</name>
-    <value>org.apache.hadoop.fs.s3a.s3guard.NullMetadataStore</value>
+    <value>org.apache.hadoop.fs.s3a.s3guard.DynamoDBMetadataStore</value>
   </property>
  <property>
     <name>fs.s3a.s3guard.ddb.region</name>
@@ -38,7 +38,7 @@
 
   <property>
     <name>hive.blobstore.optimizations.enabled</name>
-    <value>false</value>
+    <value>true</value>
   </property>
 
   <!-- Hive Configuration can either be stored in this file or in the hadoop configuration files  -->
diff --git a/itests/src/test/resources/testconfiguration.properties b/itests/src/test/resources/testconfiguration.properties
index d5ea5c1..bf48021 100644
--- a/itests/src/test/resources/testconfiguration.properties
+++ b/itests/src/test/resources/testconfiguration.properties
@@ -1072,7 +1072,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   auto_join9.q,\
   auto_join_filters.q,\
   auto_join_nulls.q,\
-  auto_smb_mapjoin_14.q,\
   auto_sortmerge_join_1.q,\
   auto_sortmerge_join_10.q,\
   auto_sortmerge_join_11.q,\
@@ -1081,7 +1080,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   auto_sortmerge_join_14.q,\
   auto_sortmerge_join_15.q,\
   auto_sortmerge_join_16.q,\
-  auto_sortmerge_join_2.q,\
   auto_sortmerge_join_3.q,\
   auto_sortmerge_join_4.q,\
   auto_sortmerge_join_5.q,\
@@ -1181,7 +1179,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   drop_partitions_filter3.q,\
   drop_partitions_ignore_protection.q,\
   dynpart_merge.q,\
-  explain_ddl.q,\
   fileformat_sequencefile.q,\
   filter_join_breaktask2.q,\
   groupby12.q,\
@@ -1212,7 +1209,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   groupby_position.q,\
   groupby_sort_1.q,\
   groupby_sort_2.q,\
-  groupby_sort_3.q,\
   groupby_sort_skew_1.q,\
   infer_bucket_sort_list_bucket.q,\
   infer_bucket_sort_multi_insert.q,\
@@ -1220,7 +1216,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   input.q,\
   input0.q,\
   input11_limit.q,\
-  input12.q,\
   input14.q,\
   input17.q,\
   input18.q,\
@@ -1245,20 +1240,13 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   input42.q,\
   input46.q,\
   input5.q,\
-  input6.q,\
-  input7.q,\
-  input8.q,\
-  input9.q,\
   input_columnarserde.q,\
-  input_dynamicserde.q,\
   input_lazyserde.q,\
   input_part10.q,\
   input_part3.q,\
-  input_part5.q,\
   input_part6.q,\
   input_part7.q,\
   input_part9.q,\
-  input_testsequencefile.q,\
   insert_into_with_schema.q,\
   join0.q,\
   join1.q,\
@@ -1274,9 +1262,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   join3.q,\
   join31.q,\
   join32.q,\
-  join34.q,\
-  join38.q,\
-  join39.q,\
   join4.q,\
   join40.q,\
   join41.q,\
@@ -1308,19 +1293,16 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   mergejoin.q,\
   mergejoins.q,\
   mergejoins_mixed.q,\
-  multiMapJoin1.q,\
   multiMapJoin2.q,\
   multi_insert_gby.q,\
   multi_insert_gby2.q,\
   multi_insert_gby3.q,\
-  multi_insert_lateral_view.q,\
   multi_insert_mixed.q,\
   multi_insert_union_src.q,\
   multi_insert_with_join.q,\
   multi_join_union.q,\
   optional_outer.q,\
   orc_create.q,\
-  orc_createas1.q,\
   outer_join_ppr.q,\
   parallel_join0.q,\
   parallel_join1.q,\
@@ -1448,7 +1430,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   skewjoin_union_remove_1.q,\
   skewjoin_union_remove_2.q,\
   smb_join_partition_key.q,\
-  smb_mapjoin9.q,\
   smb_mapjoin_1.q,\
   smb_mapjoin_10.q,\
   smb_mapjoin_11.q,\
@@ -1468,8 +1449,6 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   smb_mapjoin_3.q,\
   smb_mapjoin_4.q,\
   smb_mapjoin_5.q,\
-  smb_mapjoin_6.q,\
-  smb_mapjoin_7.q,\
   stats1.q,\
   stats10.q,\
   stats14.q,\
@@ -1516,31 +1495,22 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   union25.q,\
   union26.q,\
   union3.q,\
-  union30.q,\
   union31.q,\
   union32.q,\
-  union33.q,\
   union34.q,\
-  union4.q,\
   union5.q,\
-  union6.q,\
   union7.q,\
   union_date.q,\
   union_lateralview.q,\
   union_null.q,\
   union_remove_1.q,\
-  union_remove_10.q,\
   union_remove_2.q,\
   union_remove_20.q,\
   union_remove_3.q,\
-  union_remove_4.q,\
-  union_remove_5.q,\
   union_remove_6.q,\
   union_remove_6_subq.q,\
   union_remove_7.q,\
   union_remove_8.q,\
-  union_remove_9.q,\
-  union_top_level.q,\
   union_view.q,\
   unionall_join_nullconstant.q,\
   view.q
-- 
1.7.9.5

